### ISSUE TITLE
End-to-end verification and README

### ISSUE THREAD

**Issue by pergazuz** [pergazuz/LB-anythings#10]
## Parent

Part of #1.

## What to build

A fresh clone follows the README from nothing to a connected backend, a first Prediction, and a first automatic retrain against a real Label Studio project, using the Examples and Checkpoint migrated from the previous backend. Anything the README says is proven by doing it.

## Acceptance criteria

- [ ] The README covers: what the service is, install and run commands, the Connect Model steps with the Docker `host.docker.internal` note, the full settings table with defaults, the mapping from the old `PIPE_*` names to the `LB_*` names, the one-time migration copy of the legacy Examples and Checkpoint, and how to add a Detector behind the port.
- [ ] Following the README on a clean checkout on the target machine: the server starts, Label Studio validates, an uploaded Task shows boxes, submitting Annotations reaches the threshold and launches a run, and the next Prediction reports the new Checkpoint version.
- [ ] Start Training and mining are each exercised once per the README and behave as their tickets specify.
- [ ] Any discrepancy found between README, spec and behaviour is fixed in code or documented, never left silent.

## Blocked by

- #7
- #8
- #9

**Comment by pergazuz** [comment]
Done in 04664ee, 7ce6db2 and 463b319.

## What was verified

A ten-step pass on a `git clone` of this repo into an empty directory, following the README's
own commands with its own defaults — including the relative `LB_DATA_DIR=data`, which is what
exposed the bug below. `uv sync`, the documented migration of 333 legacy Examples and the pipe
Checkpoint, serve, Validate-and-Save, predict on a Label Studio-hosted image, an Annotation
webhook that trips the Retrain Threshold, wait for the run, predict again, Start Training, mine.

Final run, all ten steps green:

```
health: {"status":"UP","model_version":"best.pt@20260630T091311Z"}
predict: version=best.pt@20260630T091311Z boxes=3 score=0.881
webhook: HTTP 201 -> is_training: {"is_training":true}
run finished in 60s
predict after retrain: version=best.pt@20260915T030641Z boxes=1
version changed: True
Start Training: collected 5 Examples (2 unannotated, 0 unreadable); Training Run launched
mine: 153/153 frames scored, wrote 5 Hard Frames
```

199 passed / 1 skipped, mypy clean over 88 files, ruff clean, 4 import contracts kept.

## What the pass found

**The Checkpoint was written where the Detector never looks.** ultralytics resolves a relative
`project` argument against *its own* runs directory, so with the default relative
`LB_DATA_DIR=data` a Training Run wrote to `runs/detect/data/runs/active/weights/best.pt`.
`/is_training` went true then false, the run "succeeded", and predictions served the stale
Checkpoint for ever — `version changed: False`. Every earlier manual check had used an absolute
temp data directory, so nothing caught it until a clean clone ran the documented defaults.
Fixed in 04664ee by resolving `data` and `project` at the ultralytics boundary, with a
regression test; 7ce6db2 then made the runs root resolve in one place rather than two that
agreed by coincidence, and pinned the real invariant — the Checkpoint lands exactly where
`Settings.trained_checkpoint` looks — plus `exist_ok` and `workers=0`.

## What the review found that no run could

**The README never told the reader to switch on "Start model training on annotation
submission"** — the toggle the entire retrain loop depends on. The stand-in Label Studio posts
webhooks unconditionally, so it hid this: following the README to the letter, nothing would be
collected and nothing would ever retrain. The old backend documented it; this one had dropped
it. Added, along with where the Start Training button lives.

**The README claimed a guarantee the code does not give.** It said `/is_training` means a
second run cannot start on top of one going, while two sections earlier recommending
`lb-anythings train` by hand — which *is* the run body and so takes no guard. Both are now
stated honestly, and the spec records why the CLI is exempt from story 27 (463b319).

Also corrected against the code: the retrain rule is a multiple of the threshold on the
Training Set's *size*, so 333 migrated Examples means the next run is at 350, not 358; cancelled
and box-less Annotations; `LB_DATA_DIR` being read against the working directory; the
`examples/` tree as three paths; a re-runnable migration copy (verified twice over the real 333
files); `mine_with_yolo` as the second place a new Detector must be named; `pytest -m gpu`
needing the ML stack; and a Mining section, since one usage line was not enough to exercise it.

## Deviations

- **No real Label Studio is installed on this machine**, so steps 5-9 ran against a stand-in
  HTTP server that serves an uploaded frame and a project export behind a token. The wire
  protocol, media fetch, webhook ingest, retrain trigger and Start Training export are all
  exercised; what is *not* verified is the real Connect Model handshake, the real model card,
  and real payload shapes across Label Studio versions. Criterion 2 is met in substance but
  should be re-checked against a real instance before this is relied on. The README toggle
  finding above is direct evidence that a stand-in can hide this class of thing.
- **`LB_TRAIN_EPOCHS=1`** during the pass, to keep each run under a minute. The README does not
  fix epochs; nothing in the path under test depends on the count.

## Follow-up worth considering, not done here

`Settings.data_dir`, `.trained_checkpoint`, `.cache_dir` and `.mine_out` stay relative, and the
resolution now happens at the ultralytics boundary — so it depends on the working directory at
call time. Resolving them once in `Settings` with a field validator would kill the class of bug
rather than this instance. Out of scope for this ticket: it changes what the effective-settings
line logs.
